LeanplumTests/LPRegisterDeviceApiTests.m

```diff
@@ -0,0 +1,159 @@
+#import "LPTestHelper.h"
+
+@interface LPRegisterDeviceApiTests : XCTestCase
+
+@end
+
+@implementation LPRegisterDeviceApiTests
+
+- (void)setUp {
+    [super setUp];
+    [LPTestHelper setup];
+}
+
+- (void)tearDown {
+    [super tearDown];
+    [OHHTTPStubs removeAllStubs];
+}
+
+- (void)testRegisterDeviceApi {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Query timed out."];
+    [LPRegisterDeviceApi registerDevice:nil success:^ {
+        [expectation fulfill];
+    } failure:^(NSError *error) {
+    }];
+    
+    [self waitForExpectationsWithTimeout:30.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Error: %@", error);
+        }
+    }];
+}
+
+- (void)testRegisterDeviceApiWithAttributes {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Query timed out."];
+    NSDictionary *attributes = @{ LP_PARAM_EMAIL: @"test@leanplum.com" };
+    [LPRegisterDeviceApi registerDevice:attributes success:^ {
+        [expectation fulfill];
+    } failure:^(NSError *error) {
+    }];
+
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Error: %@", error);
+        }
+    }];
+}
+
+- (void)testRegisterDeviceApiWithHttpError {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Query timed out."];
+    // change device id to empty string
+    [LPTestHelper setup:APPLICATION_ID withAccessKey:DEVELOPMENT_KEY withDeviceId:@""];
+    [LPRegisterDeviceApi registerDevice:nil success:^ {
+    } failure:^(NSError *error) {
+        [expectation fulfill];
+    }];
+
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Error: %@", error);
+        }
+    }];
+}
+
+- (void)testRegisterDeviceApiWithIosError {
+    [LPTestHelper runWithApiHost:@"blah.leanplum.com" withBlock:^(void) {
+        XCTestExpectation *expectation = [self expectationWithDescription:@"Query timed out."];
+        [LPRegisterDeviceApi registerDevice:nil success:^ {
+        } failure:^(NSError *error) {
+            [expectation fulfill];
+        }];
+
+        [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+            if (error) {
+                NSLog(@"Error: %@", error);
+            }
+        }];
+    }];
+}
+
+- (void)testRegisterDeviceApiStub {
+    [LPTestHelper setupStub:200 withFileName:@"simple_success_response.json"];
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Query timed out."];
+    [LPRegisterDeviceApi registerDevice:nil success:^ {
+        [expectation fulfill];
+    } failure:^(NSError *error) {
+    }];
+
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Error: %@", error);
+        }
+    }];
+}
+
+- (void)testRegisterDeviceApiWithAttributesStub {
+    [LPTestHelper setupStub:200 withFileName:@"simple_success_response.json"];
```

Comment:
one quick note. simple_success_response.json, can we please update the name to . "simple_post_success_response.json"

Reply:
changed for simple_success and simple_error

Reply:
your awesome